pkg/sink/sink_test.go

```diff
@@ -478,3 +481,59 @@ func Test_addLabels(t *testing.T) {
 		t.Error(diff)
 	}
 }
+
+type fakeHTTPInterceptor struct {
+	i int32
+}
+
+func (f *fakeHTTPInterceptor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	data := map[string]interface{}{
+		"i": atomic.AddInt32(&f.i, 1),
+	}
+	_ = json.NewEncoder(w).Encode(data)
+}
+
+func TestExecuteInterceptor(t *testing.T) {
+	srv := httptest.NewServer(&fakeHTTPInterceptor{})
+	defer srv.Close()
+	client := srv.Client()
+	// Redirect all requests to the fake server.
+	u, _ := url.Parse(srv.URL)
+	client.Transport = &http.Transport{
+		Proxy: http.ProxyURL(u),
+	}
+
+	logger, _ := logging.NewLogger("", "")
+	r := Sink{
+		HTTPClient: srv.Client(),
+		Logger:     logger,
+	}
+
+	a := &triggersv1.EventInterceptor{
+		Webhook: &triggersv1.WebhookInterceptor{
+			ObjectRef: &corev1.ObjectReference{
+				APIVersion: "v1",
+				Kind:       "Service",
+				Name:       "foo",
+			},
+		},
+	}
+	trigger := &triggersv1.EventListenerTrigger{
+		Interceptors: []*triggersv1.EventInterceptor{a, a},
+	}
+
+	resp, err := r.executeInterceptors(trigger, httptest.NewRequest(http.MethodPost, "/", nil), nil, "", logger)
+	if err != nil {
+		t.Fatalf("executeInterceptors: %v", err)
+	}
+
+	var got map[string]int
+	if err := json.Unmarshal(resp, &got); err != nil {
+		t.Fatalf("json.Unmarshal: %v", err)
+	}
+	want := map[string]int{"i": 2}
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("-want +got: %s", diff)
+	}
+
+}
```

Comment:
nice test!!

Reply:
are there other test cases we might want to add as well, e.g. what if one interceptor fails, should the next be run?

Reply:
that particular case could maybe also be handled in a unit test for `executeInterceptors` :D